rcon/player_history.py

```diff
@@ -217,13 +232,17 @@ def ban_if_blacklisted(rcon, steam_id_64, name):
             return
 
         if player.blacklist and player.blacklist.is_blacklisted:
-            logger.warning("Player %s was banned due blacklist, reason: %s", str(
+            logger.warning("Player %s is blacklisted and has been banned, reason: %s", str(
                 player), player.blacklist.reason)
-            rcon.do_perma_ban(name, player.blacklist.reason)
             # TODO save author of blacklist
             safe_save_player_action(
-                rcon=rcon, player_name=player, action_type="PERMABAN", reason=player.blacklist.reason, by='BLACKLIST'
+                rcon=rcon, player_name=name, action_type="PERMABAN", reason=player.blacklist.reason, by='BLACKLIST'
```

Comment:
It was a mistake from me to use that function because it makes a call to `get_player_info` and the dude is already gone so it won't work. I guess it kinda works because the info is cached but that's just cheer luck. 
Since we have the  actual DB record instance in that scope `player = get_player(sess, steam_id_64)` we might as well use that directly to store the action.